src/EditorFeatures/Core.Wpf/InlineParamNameHints/InlineParamNameHintsTag.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Windows;
+using System.Windows.Controls;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Formatting;
+
+namespace Microsoft.CodeAnalysis.Editor.InlineParamNameHints
+{
+    class InlineParamNameHintsTag : IntraTextAdornmentTag
+    {
+        public readonly string TagName;
+
+        public InlineParamNameHintsTag(string text)
+            : base(CreateElement(text), null, (PositionAffinity?)PositionAffinity.Successor)
+        {
+            TagName = text;
+            if (TagName.Length != 0)
```

Comment:
When are we ever producing tags with no tag name?

Reply:
We don't, however when you delete an argument, the colon persists. This was a fix for that.

Reply:
that seems weird.  i'm not really getting teh explanation here of what's going on or why we'd need this.

Reply:
in other words, if we had a TagName of "", i'd far prefer we just not create the tag in the first place.

Reply:
@akhera99 I suspect the bug you were running into was really the bug where the tagger was throwing exceptions that got swallowed. Or as @CyrusNajmabadi points out, we should be filtering earlier.